### PR TITLE
Allow theming of OS-Viewer

### PR DESCRIPTION
**Pull request by akariv**

### CODE REVIEW

**Comment by akariv** [comment]
@pwalsh @kravets-levko please review.

See implementation details here: https://github.com/openspending/openspending/issues/1062#issuecomment-253979369

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "babbage.ui": "^1.0.0",
     "bluebird": "^3.0.5",
     "body-parser": "^1.14.1",
+    "deep-assign": "^2.0.0",
```
Comment (pwalsh) [inline package.json]: Use `_.merge` instead? we already have it.

File: app/config/themes/default.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
```
Comment (pwalsh) [inline app/config/themes/default.json:1]: @akariv 

All looks good, just to clarify:
1. The themes can't be injected from any old location - they need to actually be in the install. This seems like a good decision from a security perspective, especially as the theming API allows setting text and links, but I just want to make sure that is now the intention (not allow remote injection of theming colors, etc.)
2. Due to the complexities of theming, I guess you choose a JSON object over env vars, which seems reasonable. I just wonder if for custom installs, an env var that points to an additional theme directory, or a theme file, might also be useful, and prevent for example, docker ADD or other server actions on custom installs.

**Review by pwalsh — APPROVED** [review]

File: app/config/themes/default.json (hunk 2 above)
Comment (akariv) [reply app/config/themes/default.json:1]: 1. Yes, mostly for security (but also for performance and correctness) I opted to not process URLs via a query parameter.
2. There are 4 kinds of 'custom installs':
   
   a. Own source code & server - can just modify the JSON at will 
   b. Own server but prefer working with prebuilt docker images - ADD/COPY file is simplest (see for example [nginx Docker instructions](https://hub.docker.com/_/nginx/)). If you don't want to rebuild the image, then volume binding is also possible (which allows reusing the same host folder for all apps).
   c. 'Heroku' style deploy from git repo - in a way it's similar to (a). 
       Either way it's not much of a hassle if you only need theming, as you can fork & add a theme file using GitHub's UI alone.
   d. Partner that doesn't own code or servers - for this case you can specify a predefined theme as a query parameter.

Since all of these are relatively simple solutions, then I don't think that adding the option for using a hosted theme file will really be simpler or useful - and be worth the extra complexity and relying on an external resource on every restart.

Comment (pwalsh) [reply app/config/themes/default.json:1]: great

Comment (akariv) [reply app/config/themes/default.json:1]: (some edits above)

**Comment by akariv** [comment]
I'll wait for @kravets-levko 's opinion as well before merging

File: package.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
   "main": "index.js",
   "scripts": {
     "install": "napa 'eligrey/FileSaver.js:file-saver' 'd3/d3-plugins:d3-plugins'",
-    "postinstall": "cd ./node_modules/babbage.ui/node_modules/ && rm -rf c3 d3 bubbletree raphael pivottable/node_modules/jquery",
+    "postinstall": "echo '(cd ./node_modules/babbage.ui/node_modules/ && rm -rf c3 d3 bubbletree raphael pivottable/node_modules/jquery) || true'",
```
Comment (kravets-levko) [inline package.json]: Just a question (I don't know bash scripting so well). Why did you change this line?

Comment (akariv) [reply package.json]: Good catch... for some reason this line prevented `npm install` to work.
I'm trying a clean install now to see if it reproduces.

**Review by kravets-levko — APPROVED** [review]
